great_expectations/expectations/window.py

```diff
@@ -0,0 +1,21 @@
+from great_expectations.compatibility import pydantic
+
+
+class Offset(pydantic.BaseModel):
+    """
+    A threshold in which a metric will be considered passable
+    """
+
+    positive: float
+    negative: float
+
+
+class Window(pydantic.BaseModel):
+    """
+    A definition for a temporal window across <`range`> number of previous invocations
+    """
+
+    constraint_fn: str
+    parameter_name: str
+    range: int
+    offset: Offset
```

Comment:
I would suggest we set the `extra` ignore behavior to forbid by default.
Otherwise it can be easy to start providing extra arguments to this class without noticing.

>whether to ignore, allow, or forbid extra attributes during model initialization. Accepts the string values of 'ignore', 'allow', or 'forbid', or values of the Extra enum (default: Extra.ignore). 'forbid' will cause validation to fail if extra attributes are included, 'ignore' will silently ignore any extra attributes, and 'allow' will assign the attributes to the model.


https://docs.pydantic.dev/1.10/usage/model_config/#options

Reply:
Yep great call, added in a63abe335, tweaked imports in 24edd2f2aa8042e97a69c246f74495fbc7c57bc0